queuelib/pqueue.py

```diff
@@ -22,24 +25,24 @@ class PriorityQueue:
 
     """
 
-    def __init__(self, qfactory, startprios=()):
+    def __init__(self, qfactory: Callable, startprios: Iterable[int] = ()) -> None:
```

Comment:
While `qfactory` is indeed a callable, it should not just be any callable, it needs a queue signature (e.g. `push`, `pop`). I wonder if we can be more explicit about that.

Reply:
What about 9cf44b5c3b9a0abb7a4d7f1452dd6bcf22b4dc3a? Is it too hacky?